mail_tracking/models/mail_thread.py

```diff
@@ -29,17 +29,24 @@ def _get_failed_message_domain(self):
             ("mail_tracking_ids.state", "in", list(failed_states)),
         ]
 
-    @api.returns("self", lambda value: value.id)
-    def message_post(self, *args, **kwargs):
+    @api.model
+    def _message_route_process(self, message, message_dict, routes):
         """Adds CC recipient to the message.
 
         Because Odoo implementation avoid store 'from, to, cc' recipients we
         ensure that this information its written into the mail.message record.
         """
-        kwargs.update(
-            {"email_cc": kwargs.get("cc", False), "email_to": kwargs.get("to", False)}
+        thread_id = super()._message_route_process(message, message_dict, routes)
+        message = self.env["mail.message"].search(
+            [("message_id", "=", message_dict.get("message_id"))]
+        )
+        message.write(
```

Comment:
Can we avoid this double writing? This one is critical due to the high volume of the messages, so we shouldn't write several times and do it at message creation.